### PR TITLE
feat(storage): add redis storage

### PR DESCRIPTION
**Pull request by eladav**
* re-organize types
* add composite storage
Solves #48 and #17

### CODE REVIEW

File: server/CompositeStorage/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  ComponentsStorage,
+  Component,
+  File,
+  Index,
+  ComponentTree,
+  ComponentGetter,
+  Maybe
+} from '@dynamico/driver';
+
+export class CompositeStorage implements Storage {
```
Comment (EladBezalel) [inline server/CompositeStorage/lib/index.ts]: I think it's a composition storage
```suggestion
export class CompositionStorage implements Storage {
```

File: server/CompositeStorage/lib/index.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  Storage,
+  IndexStorage,
+  ComponentsStorage,
+  Component,
+  File,
+  Index,
+  ComponentTree,
+  ComponentGetter,
+  Maybe
+} from '@dynamico/driver';
```
Comment (EladBezalel) [inline server/CompositeStorage/lib/index.ts]: Why are we getting these types out of the driver and not the common types?

File: common/types/index.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  upsertIndex(index: Index): Promise<void>;
+}
+
+export interface ComponentsStorage {
+  getComponentTree(): Promise<ComponentTree>;
+  getComponent(name: string, version: string): Promise<Maybe<ComponentGetter>>;
+  saveComponent(component: Component, files: File[]): Promise<void>;
+}
+
+export type Storage = IndexStorage & ComponentsStorage;
```
Comment (EladBezalel) [inline common/types/index.ts]: beautiful

File: server/CompositeStorage/lib/index.ts (hunk 2 above)
Comment (eladav) [reply server/CompositeStorage/lib/index.ts]: Yeah it's kind of weird... Not sure what's the correct structure here is but I wanted some package to export the types and not publish a `@types/dynamico` package. I'll play with it some more to find the best setup...

**Pull request merged**